fcsparser/api.py

```diff
@@ -193,6 +193,31 @@ def read_header(self, file_handle, nextdata_offset=0):
 
         self.annotation['__header__'] = header
 
+    def _extract_text_dict(self, raw_text):
+        """Parse the TEXT segment of the FCS file into a python dictionary."""
+        delimiter = raw_text[0]
+
+        if raw_text[-1] != delimiter:
+            raw_text = raw_text.strip()
+            if raw_text[-1] != delimiter:
+                msg = (u'The first two characters were:\n {}. The last two characters were: {}\n'
+                       u'Parser expects the same delimiter character in beginning '
+                       u'and end of TEXT segment'.format(raw_text[:2], raw_text[-2:]))
+                raise ParserFeatureNotImplementedError(msg)
+
+        # Below 1:-1 used to remove first and last characters which should be reserved for delimiter
+        # Also, the delimiter is "quoted" by being repeated (two consecutive delimiters)
```

Comment:
Is this part of the official fcs specification?

Could you paraphrase the comment to say that it is escaped by being repeated (rather than quoted)... that's the usual terminology for an escape character. 

---

Would you be able to rephrase this code to be more explicit / direct about what it is doing?

Instead of going through an intermediate array (`delimiter_split_arr`), perhaps we could directly identify the places to split? Any chance there's a simple regexp that could be used?

(the logic is quite difficult to follow)

Reply:
Yes, it's been in since at least FCS 2.0 (see the links in the PR). They use the term "quoted", but I agree that "escaped" is more appropriate.

I was able to come up with a regexp which makes the process arguably more straightforward. I think it's still a little hairy, though. It needs both a look-ahead and a look-behind, which unfortunately prevent using `re.split()` (see the code comments I added).